bitnami/rabbitmq/templates/_helpers.tpl

```diff
@@ -38,15 +38,52 @@ Return the proper Docker Image Registry Secret Names
 {{- end -}}
 
 {{/*
-Get the password secret.
+Returns the available value for certain key in an existing secret (if it exists),
+otherwise it generates a random value.
 */}}
-{{- define "rabbitmq.secretPasswordName" -}}
-    {{- if .Values.auth.existingPasswordSecret -}}
-        {{- printf "%s" (tpl .Values.auth.existingPasswordSecret $) -}}
+{{- define "getValueFromSecret" }}
+    {{- $len := (default 16 .Length) | int -}}
+    {{- $obj := (lookup "v1" "Secret" .Namespace .Name).data -}}
+    {{- if $obj }}
+        {{- index $obj .Key | trimAll "\"" | b64dec -}}
```

Comment:
this is a change from the previous function; added a pipe to trim " as this was implemented in secrets.yaml:9; I believe to strip " from passwords. Its nice to keep if someone accidently adds quotations around a password.